.editorconfig

```diff
@@ -3,9 +3,11 @@ root = true
 [*]
 charset = utf-8
 indent_style = space
-indent_size = 2
 end_of_line = lf
 insert_final_newline = true
+
+[*.{!md}]
+indent_size = 2
 trim_trailing_whitespace = true
```

Comment:
why do you want these exceptions for Markdown?

also this change is not related to Windows fixes, so I would expect to see them in a different commit with explanation in a commit message somewhere

Reply:
Previously we specified the file extensions to run eclint on. In the new script, you don't specify it, so it would check markdown as well and complain quite a bit about it.

In markdown, you don't want these rules:
- If you specify indent_size = 2 for markdown, you won't be able to do codeblocks that have comment blocks inside like this:
```js
/**
 * <-- Indent of 1 instead of 2, eclint will complain
 */
const myJsVar = 'foo';
```
- The reason we don't want to trim trailing whitespace in markdown, I forgot what it was exactly but there was a very specific reason at some point where we needed to have a space and it was auto trimmed wrongly. @daKmoR  I believe you may remember what this was, is this still relevant now?

Reply:
So yeah there is *some* relevance but I do agree having it in two separate commits would be better so I can always change that once we agree how to proceed here

Reply:
first point is applicable to all files containing such multiline commets, this is why we have this exception

```
[*.{html,js}]
block_comment_start = /**
block_comment = *
block_comment_end = */
```

so instead of disabling this rule completely for Markdown, I would just add `.md` to this last config with `html` and `js` extensions

Reply:
regarding the second point: never encountered any need to have explicit space in the end of any line in Markdown, but has encountered hundred times how people accidentally left space or multiple spaces there for no reason, so unless @daKmoR has a good reason, I would rather not do it

at the same time, if we include prettier by default for Markdown, all these problems will be gone, since prettier has very nice formatting rules for Markdown which also fix this sort of problems like trailing space

Reply:
so for now let's focus on Windows fixes and leave Markdown changes for next time (next MR maybe where we can continue discussing how to do it properly?)

Reply:
The thing is, the eclint script changed to check all file extensions instead of just .html, .js and .css

So we can't really split it in another MR because this MR will not pass the build test for eclint without these markdown changes. 

![image](https://user-images.githubusercontent.com/36734656/58864160-867bb400-86b4-11e9-84e7-c575757728f4.png)

Reply:
Myeah, I checked where we have trailing whitespace in md (only 1 file) and it really makes no difference, so I removed it 👍

See latest version of the changes, should address all your concerns, apart from the splitting it up in 2 MRs